astropy/cosmology/core.py

```diff
@@ -1060,10 +1055,11 @@ def comoving_distance(self, z):
         """
 
         from scipy.integrate import quad
+        inv_e = self.inv_efunc
         if not isiterable(z):
```

Comment:
This stanza is used many times. Maybe factor it out?
```
return self._hubble_distance * _vectorize_if_needed(lambda _z: quad(inv_e, 0, _z)[0])(z)
```
This would make the code more readable and allow one to experiment with faster implementations more easily (e.g., `np.vectorize?` says: """ The `vectorize` function is provided primarily for convenience, not for
performance. The implementation is essentially a for loop."""